tests/unit/test_decoder.py

```diff
@@ -93,7 +93,8 @@ def test_blueprint_weights(self, enc_state):
 class TestDecoder:
     """A class that defines tests for the CV and MWPM decoder."""
 
-    def test_CV_decoder(self, enc_state):
+    @classmethod
```

Comment:
What is the purpose of @classmethod?

Reply:
Ah, yes, this is quite important and was a common mistake for us and Python developers in general:

One should not create a method (with self) in a class that doesn't use any instance variables. It's quite non-pythonic for good reasons. In such cases, we should use @classmethod or @staticmethod as I updated.

Reply:
Understood - but does this imply that these should have the @staticmethod decorator, since we never have to refer to the class?

Also, does this interfere in any way with how pytest works?

Reply:
> Also, does this interfere in any way with how pytest works?

Be aware that using the `@classmethod` decorator with pytest fixtures is not supported and leads to unexpected errors. There's an [old thread](https://github.com/pytest-dev/pytest/pull/3781#issue-347639817) on the `pytest-dev` repo about this issue. I have also seen weird behavior when using this decorator on `test_*` functions. The issues potentially stems from the fact that class-methods get the class itself as a parameter (`cls`) instead of an instance of the class (`self`), interfering with the inner workings of pytest.

My suggestion is to use the signature `test_*(self, args...)` for test methods. Helper methods called within test methods can use the `@classmethod` decorator.

Reply:
It's important to note here that PyTest doesn't always work in obvious (pythonic) ways, meaning that, as Sebas pointed out, one should be careful when marking test functions with non-pytest decorators.

> I think we should enclose expressions with @ in angle brackets, otherwise random GitHub users get pinged. :D
I love that there are users named `classmethod` and `staticmethod`. 😂

Reply:
Thanks all, as agreed I will remove non-pytest decorators from all test files and add `pylint: disable` as needed for now.